File: 백준/Silver/10866. 덱/덱.java
import java.io.IOException;
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.StringTokenizer;
import java.util.LinkedList;



public class Main{
    public static void main(String [] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        StringBuilder sb = new StringBuilder();

        int n = Integer.parseInt(st.nextToken());

        LinkedList<Integer> deque = new LinkedList<>();

        while (n-- > 0) {
            st = new StringTokenizer(br.readLine());
            String commend = st.nextToken();
            switch (commend) {
                case "push_front":
                    deque.addFirst(Integer.parseInt(st.nextToken()));
                    continue;
                case "push_back":
                    deque.add(Integer.parseInt(st.nextToken()));
                    continue;

                case "pop_front":
                    if (!deque.isEmpty()) {
                        sb.append(deque.pollFirst()).append("\n");
                        continue;
                    }
                    else {
                        sb.append("-1").append("\n");
                        continue;
                    }
                case "pop_back":
                    if (!deque.isEmpty()) {
                        sb.append(deque.pollLast()).append("\n");
                        continue;
                    }
                    else {
                        sb.append("-1").append("\n");
                        continue;
                    }

                case "size":
                    sb.append(deque.size()).append("\n");
                    continue;

                case "empty":
                    if (deque.isEmpty()) {
                        sb.append(1).append("\n");
                        continue;
                    }
                    else {
                        sb.append(0).append("\n");
                        continue;
                    }
                case "front":
                    if (deque.isEmpty()) {
                        sb.append("-1").append("\n");
                        continue;
                    }
                    else {
                        sb.append(deque.peekFirst()).append("\n");
                        continue;
                    }

                case "back":
                    if (deque.isEmpty()) {
                        sb.append("-1").append("\n");
                        continue;
                    }
                    else {
                        sb.append(deque.peekLast()).append("\n");
                        continue;
                    }
            }
        }

        System.out.println(sb);


    }
}

File: 백준/Silver/10866. 덱/덱.py
import sys
input = sys.stdin.readline

N = int(input())
deck = []

for i in range(N):

    n = input().split()
    if n[0] == 'push_front':
        deck.insert(0, n[1])
    elif n[0] == 'push_back':
        deck.append(n[1])
    elif n[0] == 'front':
        if deck:
            print(deck[0])
        else:
            print('-1')
    elif n[0] == 'back':
        if deck:
            print(deck[-1])
        else:
            print('-1')

    elif n[0] == 'size':
        print(len(deck))

    elif n[0] == 'empty':
        if deck:
            print('0')
        else:
            print('1')

    elif n[0] == 'pop_front':
        if deck:
            print(deck.pop(0))
        else:
            print('-1')
    elif n[0] == 'pop_back':
        if deck:
            print(deck.pop(-1))
        else:
            print('-1')

